pkg/backup/backup.go

```diff
@@ -191,77 +208,74 @@ func (b *Backup) saveSnap(lastSnapRev int64) (int64, error) {
 	if len(pods) == 0 {
 		msg := "no running etcd pods found"
 		logrus.Warning(msg)
-		return lastSnapRev, fmt.Errorf(msg)
+		return nil, fmt.Errorf(msg)
 	}
-	member, rev := getMemberWithMaxRev(pods, b.etcdTLSConfig, b.selfHosted)
+	member, rev := getMemberWithMaxRev(pods, bm.etcdTLSConfig)
 	if member == nil {
 		logrus.Warning("no reachable member")
-		return lastSnapRev, fmt.Errorf("no reachable member")
+		return nil, fmt.Errorf("no reachable member")
 	}
 
 	if rev <= lastSnapRev {
 		logrus.Info("skipped creating new backup: no change since last time")
-		return lastSnapRev, nil
+		return nil, nil
```

Comment:
@xiang90 I am unsure what to return here. I return `nil,nil` because no backup is performed and skipping is not considered an error by judgement of previous code.